Fix above/below counting.

postwait opened the pull request:
* Make below inclusive (no change)
 * Make above exclusive (change)
 * Add explicit `_{in,ex}clusive` suffixes for interested parties.

bdunavant commented:
In relation to issue #72

pradeepcrao commented on src/circllhist.c:
```diff
@@ -705,7 +708,29 @@ hist_approx_count_below(const histogram_t *hist, double threshold) {
+    if(bucket_bound < 0.0)
+      bucket_upper = bucket_bound;
+    else
+      bucket_upper = bucket_bound + hist_bucket_to_double_bin_width(hist->bvs[i].bucket);
```
nit: This doesn't change the answer for `hist_approx_count_below_exclusive` but for `abs(bucket_bound) < 10`, the value returned by `hist_bucket_to_double_bin_width` is actually `width/10` as it doesn't account for lines 971 through 973 in `int_scale_to_hist_bucket`. Don't know if this is intentional.

postwait replied:
This sounds like a different bug, can you describe in more detail what you're seeing?  The internal encoding represents 1.0 to 9.9 as 10 to 99.  So the "width" [encoded 10, encoded 11) is 1 representing the 0.1 between [1.0, 1.1)

postwait replied:
I did see the inequality was off in below_exlcusive and the bucket_upper is equal to the lower bound of the greater bucket and we should be looking for things less than that, not less than or equal to.

pradeepcrao replied:
Your comment about internal encoding makes sense. Thank you for clearing that up.

Regarding the inequality, I believe what you had before was correct, unless I am misunderstanding something.

pradeepcrao commented on src/circllhist.c:
```diff
@@ -721,7 +721,7 @@ hist_approx_count_below_exclusive(const histogram_t *hist, double threshold) {
       bucket_upper = bucket_bound;
     else
       bucket_upper = bucket_bound + hist_bucket_to_double_bin_width(hist->bvs[i].bucket);
-    if(bucket_upper <= threshold)
```
Since the buckets are defined as `[bucket_lower, bucket_upper)`,  (for values  > 0) I believe what you had earlier was correct. i.e. this needs to be `bucket_upper <=threshold`. Since all values in the bucket are strictly < threshold if `bucket_upper<=threshold`.

pradeepcrao replied:
For eg, if `threshold == 9`, we want the count for the bucket `[9, 10)` to be included for this method.

postwait replied:
The bucket is [9.0, 9.1), but yes.  Did the last commit fix that?  The bin width should be 0.1 there.  In below_excusive, we would not want it counted.  In below_inclusive we would want that counted.

postwait replied:
Yikes, I think inclusive and exclusive are backwards in my patch.

postwait commented:
@pradeepcrao I've refactored this entirely to use bin comparisons which should (1) be significantly faster and (2) be much easier to read.  Also, the tests clearly show that below and above counts are inclusive by default so I've changed that as well.

pradeepcrao reviewed (approved):
Looks good to me. Thanks so much!
